### DISCUSSION TITLE
agent in py-uvm

### DISCUSSION THREAD

**Discussion by ark0100** [raysalemi/Python4RTLVerification#9] [Q&A]
Hello,

I am trying to implement uvm agents concept in one of my design framework .I have made separate classes for different agents and trying to override it by using uvm factory concepts.
The issue I am facing is that it gets override successfully when i use it under pyuvm.test decorator.  But when i tried to do the same thing outside the decorator it never gets override .

Thanks in advance and looking forward for your response.

**Comment by raysalemi** [comment]
I don't understand your question based on this description.  Could you
share some code?
